### ISSUE TITLE
'SparseTensor' object is not callable

### ISSUE THREAD

**Issue by jbarth-ubhd** [Calamari-OCR/calamari#192]
I'm trying to train on lines (with random characters) created by Imagemagick (font).
Error Message:

```
==================================================================================================
Total params: 1,550,998
Trainable params: 1,550,998
Non-trainable params: 0
__________________________________________________________________________________________________
None
Creating initial network configuration as current best.
Storing checkpoint to '/dwork/ocrImage/_jb/bembo/best.ckpt'
/home/jb/calamari_venv/lib/python3.6/site-packages/tensorflow/python/keras/engine/functional.py:543: UserWarning: Input dict contained keys ['input_data_params'] which did not match any model input. They will be ignored by the model.
  [n for n in tensors.keys() if n not in ref_input_names])
  Storing checkpoint to '/dwork/ocrImage/_jb/bembo/model_00000500.ckpt'
Storing checkpoint to '/dwork/ocrImage/_jb/bembo/model_00001000.ckpt'
Traceback (most recent call last):
  File "/home/jb/calamari_venv/bin/calamari-train", line 8, in <module>
    sys.exit(main())
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/calamari_ocr/scripts/train.py", line 367, in main
    run(args)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/calamari_ocr/scripts/train.py", line 359, in run
    progress_bar=not args.no_progress_bars
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/calamari_ocr/ocr/trainer.py", line 197, in train
    self._run_train(train_net, train_start_time, progress_bar, self.dataset, self.validation_dataset, training_callback)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/calamari_ocr/ocr/trainer.py", line 213, in _run_train
    train_net.train(train_dataset, val_dataset, checkpoint_params, self.txt_postproc, progress_bar, training_callback)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/calamari_ocr/ocr/backends/tensorflow_backend/tensorflow_model.py", line 332, in train
    v_cb, es_cb
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/tensorflow/python/keras/engine/training.py", line 108, in _method_wrapper
    return method(self, *args, **kwargs)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/tensorflow/python/keras/engine/training.py", line 1103, in fit
    callbacks.on_train_batch_end(end_step, logs)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/tensorflow/python/keras/callbacks.py", line 440, in on_train_batch_end
    self._call_batch_hook(ModeKeys.TRAIN, 'end', batch, logs=logs)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/tensorflow/python/keras/callbacks.py", line 289, in _call_batch_hook
    self._call_batch_end_hook(mode, batch, logs)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/tensorflow/python/keras/callbacks.py", line 309, in _call_batch_end_hook
    self._call_batch_hook_helper(hook_name, batch, logs)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/tensorflow/python/keras/callbacks.py", line 346, in _call_batch_hook_helper
    hook(batch, numpy_logs)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/tensorflow/python/keras/callbacks.py", line 687, in on_train_batch_end
    self.on_batch_end(batch, logs=logs)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/calamari_ocr/ocr/backends/tensorflow_backend/callbacks/visualize.py", line 52, in on_batch_end
    cer, target, decoded = self._generate(1)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/calamari_ocr/ocr/backends/tensorflow_backend/callbacks/visualize.py", line 67, in _generate
    cer, target, decoded = zip(*[self.predict_func(next(it)) for _ in range(count)])
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/calamari_ocr/ocr/backends/tensorflow_backend/callbacks/visualize.py", line 67, in <listcomp>
    cer, target, decoded = zip(*[self.predict_func(next(it)) for _ in range(count)])
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/tensorflow/python/keras/backend.py", line 3940, in func
    outs = model(model_inputs)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/tensorflow/python/keras/engine/base_layer.py", line 985, in __call__
    outputs = call_fn(inputs, *args, **kwargs)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/tensorflow/python/keras/engine/functional.py", line 386, in call
    inputs, training=training, mask=mask)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/tensorflow/python/keras/engine/functional.py", line 508, in _run_internal_graph
    outputs = node.layer(*args, **kwargs)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/tensorflow/python/keras/engine/base_layer.py", line 985, in __call__
    outputs = call_fn(inputs, *args, **kwargs)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/tensorflow/python/keras/layers/core.py", line 903, in call
    result = self.function(inputs, **kwargs)
  File "/home/jb/calamari_venv/lib/python3.6/site-packages/calamari_ocr/ocr/backends/tensorflow_backend/tensorflow_model.py", line 183, in <lambda>
    sparse_decoded = KL.Lambda(lambda args: sparse_decoded(*args), name='sparse_decoded')(
TypeError: 'SparseTensor' object is not callable
2020-12-14 20:02:09.095670: W tensorflow/core/kernels/data/generator_dataset_op.cc:103] Error occurred when finalizing GeneratorDataset iterator: Failed precondition: Python interpreter state is not initialized. The process may be terminated.
	 [[{{node PyFunc}}]]
2020-12-14 20:02:09.098195: W tensorflow/core/kernels/data/generator_dataset_op.cc:103] Error occurred when finalizing GeneratorDataset iterator: Failed precondition: Python interpreter state is not initialized. The process may be terminated.
	 [[{{node PyFunc}}]]
```

**Comment by maxnth** [comment]
AFAIK this is caused by running calamari 1.0.5 (or any other release available on PyPI) with tensorflow>=2.3.
Installing calamari from source should fix this as @andbue already merged a fix for this (#184) into the master with a8614b5323ecf038511df15669357045f5d640c4.

**Comment by jbarth-ubhd** [comment]
Thanks, Tensorflow 2.2.1 works.
